sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -218,6 +226,77 @@ case class GetArrayStructFields(
   }
 }
 
+object ExtractNestedArray {
+  type ReturnType = Option[(DataType, Boolean, Seq[Boolean])]
+
+  def unapply(dataType: DataType): ReturnType = {
+    extractArrayType(dataType)
+  }
+
+  def extractArrayType(dataType: DataType): ReturnType = {
```

Comment:
@guiyanakuang can you address the comments in https://github.com/apache/spark/pull/28860?

Reply:
Yes, I will modify some of the code and supplementary documentation later.